File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book.sort()  # 전화번호를 사전순으로 정렬하여 비교를 쉽게 함
    
    for i in range(len(phone_book) - 1):
        if phone_book[i] == phone_book[i + 1][:len(phone_book[i])]:
            return False
    
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    var answer = true;
    phone_book.sort()
    for (let i = 1; i<phone_book.length; i++ ){
        if (phone_book[i].startsWith(phone_book[i-1])) return false 
    }
    return answer;
}